addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -244,9 +244,54 @@ def action_send_sms(self, res_ids=None):
 
             composer = self.env['sms.composer'].with_context(active_id=False).create(mailing._send_sms_get_composer_values(res_ids))
             composer._action_send_sms()
-            mailing.write({'state': 'done', 'sent_date': fields.Datetime.now()})
+            mailing.write({
+                'state': 'done',
+                'sent_date': fields.Datetime.now(),
+                'kpi_mail_required': not mailing.sent_date,
+                })
         return True
 
+    # ------------------------------------------------------
+    # STATISTICS
+    # ------------------------------------------------------
+    def _prepare_statistics_email_values(self):
+        """Return some statistics that will be displayed in the mailing statistics email.
+
+        Each item in the returned list will be displayed as a table, with a title and
+        1, 2 or 3 columns.
+        """
+        values = super(Mailing, self)._prepare_statistics_email_values()
+        if self.mailing_type == 'sms':
+            mailing_type = self._get_pretty_mailing_type()
+            values['title'] = _('24H Stats of %s "%s"') % (mailing_type, self.subject)
+            values['kpi_data'][0] = {
+                'kpi_fullname': _('Report for %i %s Sent') % (self._get_count_sent(), mailing_type),
+                'kpi_col1': {
+                    'value': f'{self.received_ratio}%',
+                    'col_subtitle': '%s (%i)' % (_('RECEIVED'), self.delivered),
+                },
+                'kpi_col2': {
+                    'value': f'{self.clicks_ratio}%',
+                    'col_subtitle': '%s (%i)' % (_('CLICKED'), self.clicked),
+                },
+                'kpi_col3': {
+                    'value': f'{self.bounced_ratio}%',
+                    'col_subtitle': '%s (%i)' % (_('BOUNCED'), self.bounced),
+                },
+                'kpi_action': None,
+            }
+        return values
+
+    def _get_pretty_mailing_type(self):
+        if self.mailing_type == 'sms':
+            return _('SMS Text Message')
+        return super(Mailing, self)._get_pretty_mailing_type()
+
+    def _get_count_sent(self):
+        if self.mailing_type == 'sms':
+            return self.expected
```

Comment:
Why are you using expected and not sent ?

Reply:
In the case of SMS, I think that sent contains the number of sms well received. During my tests, I sent 3 sms, 1 of which was not received correctly (false num). sent = 2 and expected = 3. And all kpi in % was based on the value 3. So I had in my email 66% of sms received on 2 sent. Which is false.